tests/test_vnet.py

```diff
@@ -0,0 +1,705 @@
+        asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+        app_db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+
+        # FIXME - This entry must be created by VRFMgr
+        create_entry_pst(
+            app_db,
+            "VXLAN_TUNNEL_MAP", ':', "%s:%s" % (tunnel_name, vnet_name),
+            [
+                ("vni", vni_id),
+                ("vrf", vnet_name),
+            ],
+        )
+
+        time.sleep(2)
+
+        if (self.tunnel_map_map.get(tunnel_name) is None):
+            tunnel_map_id = get_created_entries(asic_db, self.ASIC_TUNNEL_MAP, self.tunnel_map_ids, 2)
+        else:
+            tunnel_map_id = self.tunnel_map_map[tunnel_name]
+
+        tunnel_map_entry_id = get_created_entries(asic_db, self.ASIC_TUNNEL_MAP_ENTRY, self.tunnel_map_entry_ids, 2)
+
+        # check that the vxlan tunnel termination are there
+        assert how_many_entries_exist(asic_db, self.ASIC_TUNNEL_MAP_ENTRY) == (len(self.tunnel_map_entry_ids) + 2), "The TUNNEL_MAP_ENTRY is created too early"
+
+        check_object(asic_db, self.ASIC_TUNNEL_MAP_ENTRY, tunnel_map_entry_id[0],
+            {
+                'SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP_TYPE': 'SAI_TUNNEL_MAP_TYPE_VIRTUAL_ROUTER_ID_TO_VNI',
+                'SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP': tunnel_map_id[1],
+                'SAI_TUNNEL_MAP_ENTRY_ATTR_VIRTUAL_ROUTER_ID_KEY': vr_ids[0],
+                'SAI_TUNNEL_MAP_ENTRY_ATTR_VNI_ID_VALUE': vni_id,
+            }
+        )
+
+        check_object(asic_db, self.ASIC_TUNNEL_MAP_ENTRY, tunnel_map_entry_id[1],
+            {
+                'SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP_TYPE': 'SAI_TUNNEL_MAP_TYPE_VNI_TO_VIRTUAL_ROUTER_ID',
+                'SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP': tunnel_map_id[0],
+                'SAI_TUNNEL_MAP_ENTRY_ATTR_VNI_ID_KEY': vni_id,
+                'SAI_TUNNEL_MAP_ENTRY_ATTR_VIRTUAL_ROUTER_ID_VALUE': vr_ids[1],
+            }
+        )
+
+        self.tunnel_map_entry_ids.update(tunnel_map_entry_id)
+
+
+class TestVnetOrch(VnetVxlanTunnel):
+
+    '''
+    Test 1 - Create Vlan Interface, Tunnel and Vnet
+    '''
+    def test_vnet_orch_1(self, dvs, testlog):
+        tunnel_name = 'tunnel_1'
+        vr_set = set()
+        self.fetch_exist_entries(dvs)
+
+        create_vxlan_tunnel(dvs, tunnel_name, '10.10.10.10')
+
+        vr_ids = create_vnet_entry(dvs, 'Vnet_2000', tunnel_name, '2000', "", self.vnet_vr_ids)
+        self.vnet_vr_ids.update(vr_ids)
+
+        self.check_vxlan_tunnel_entry(dvs, tunnel_name, 'Vnet_2000', '2000', vr_ids)
+
+        tun_id = self.check_vxlan_tunnel(dvs, tunnel_name, '10.10.10.10')
+
+        create_vlan_interface(dvs, "Vlan100", self.vlan_ids, "Ethernet24", "Vnet_2000", "100.100.3.1/24", vr_ids[0])
+
+        create_vlan_interface(dvs, "Vlan101", self.vlan_ids, "Ethernet28", "Vnet_2000", "100.100.4.1/24", vr_ids[0])
+
+        get_vr_set(vr_set, [vr_ids])
```

Comment:
it is not intuitive of this get_vr_set function. can you try to remove it and pass the vnet_vrids into create_vnet_routes()? there, you know whether it is ingress or egress route and you can check inside the create_vnet_routes()?